packages/matter.js/src/cluster/Cluster.ts

```diff
@@ -22,58 +22,487 @@ export const enum AccessLevel {
+    writable: false,
+    fixed: true,
+    scene,
+    persistent,
+    fabricScoped: false,
+    omitChanges,
+    default: conformanceValue,
+    readAcl
+});
+
+export const ConditionalFixedAttribute = <T, V extends T, F extends BitSchema>(id: number, schema: TlvSchema<T>, {
+    scene = false,
+    persistent = false,
+    omitChanges = false,
+    default: conformanceValue,
+    readAcl = AccessLevel.View,
+    optionalIf = [],
+    mandatoryIf = [],
+}: ConditionalAttributeOptions<V, F>): ConditionalFixedAttribute<T, F> => ({
+    id,
+    schema,
+    optional: true,
+    optionalIf,
+    mandatoryIf,
+    writable: false,
+    fixed: true,
+    scene,
+    persistent,
+    fabricScoped: false,
+    omitChanges,
+    default: conformanceValue,
+    readAcl
+});
 
 export type MandatoryAttributeNames<A extends Attributes> = { [K in keyof A]: A[K] extends OptionalAttribute<any> ? never : K }[keyof A];
 export type OptionalAttributeNames<A extends Attributes> = { [K in keyof A]: A[K] extends OptionalAttribute<any> ? K : never }[keyof A];
 
 /* Interfaces and helper methods to define a cluster command */
 export const TlvNoResponse = TlvVoid;
-export interface Command<RequestT, ResponseT> { optional: boolean, requestId: number, requestSchema: TlvSchema<RequestT>, responseId: number, responseSchema: TlvSchema<ResponseT> }
-export interface OptionalCommand<RequestT, ResponseT> extends Command<RequestT, ResponseT> { optional: true }
+
+export interface Command<RequestT, ResponseT> {
+    optional: boolean,
+    requestId: number,
+    requestSchema: TlvSchema<RequestT>,
+    responseId: number,
+    responseSchema: TlvSchema<ResponseT>
+}
+
+export interface OptionalCommand<RequestT, ResponseT> extends Command<RequestT, ResponseT> {
+    optional: true,
+}
+
+export interface ConditionalCommand<RequestT, ResponseT, F extends BitSchema> extends OptionalCommand<RequestT, ResponseT> {
+    mandatoryIf?: ConditionalFeatureList<F>,
+    optionalIf?: ConditionalFeatureList<F>
+}
+
 export type ResponseType<T extends Command<any, any>> = T extends OptionalCommand<any, infer ResponseT> ? ResponseT : (T extends Command<any, infer ResponseT> ? ResponseT : never);
 export type RequestType<T extends Command<any, any>> = T extends OptionalCommand<infer RequestT, any> ? RequestT : (T extends Command<infer RequestT, any> ? RequestT : never);
 
-export const Command = <RequestT, ResponseT>(requestId: number, requestSchema: TlvSchema<RequestT>, responseId: number, responseSchema: TlvSchema<ResponseT>): Command<RequestT, ResponseT> => ({ optional: false, requestId, requestSchema, responseId, responseSchema });
-export const OptionalCommand = <RequestT, ResponseT>(requestId: number, requestSchema: TlvSchema<RequestT>, responseId: number, responseSchema: TlvSchema<ResponseT>): OptionalCommand<RequestT, ResponseT> => ({ optional: true, requestId, requestSchema, responseId, responseSchema });
+interface ConditionalCommandOptions<F extends BitSchema> {
```

Comment:
Why did you define a different interface for every element type?  Vs. just ElementOptions

Reply:
In attribute it is "special" anyway. For commands and events yes could be reused .. decided to kep separated for later extensibility (else we need split up later) ...

Reply:
No reason you can't specialize the base interface if necessary.  So might be ElementOptions -> AttributeOptions -> ConditionalAttribute but only ElementOptions -> ConditionalEvent.  NBD as is but expect me to squawk any time lines are duplicated in four places. 🙂